README.md

```diff
@@ -121,3 +121,46 @@ You can also pass nativefier flags, and mount additional volumes to provide loca
 ## License
 
 [MIT](LICENSE.md)
+
+## TS: Minimum to release beta
+
+- [x] Move cli to TS
+- [x] Move app to TS
+- [x] Package.json nits: move away from "^A.B.C" to more understandable "A.x
+- [x] Upgrade electron-packager and other deps
+- [x] Fix node_modules badly pruned when copied to app
+- [x] Get rid of Babel / webpack
+- [x] Move from Gulp to simple npm scripts
+- [x] Replace quirky & broken `progress` with basic logging and restore electron-packager logging
+- [x] Replace `async` module + callbacks with real native async/await
+- [x] Cleanup/port old npm tasks/scripts
+- [x] Make exports more idiomatic without default & index.ts crap
+- [x] Get rid of micro-packages easily inlined
+- [x] Cleanup pass on cli/app code to make more modern async TS idiomatic
+- [x] More verbose & normal logging everywhere
+- [x] Disable logging by default in tests & add tasks to enable at any level.
+- [x] Fix icon not showing up on Linux
+- [x] Review .npmignore and published zip. Test installing it.
+- [ ] Stopping to webpack `app` template caused https://github.com/npm/cli/issues/757 .
+      Find a workaround or revert to webpack.
```

Comment:
I think I identified the issue for this one here: https://github.com/npm/cli/issues/757#issuecomment-581773898

Reply:
> I think I identified the issue for this one here: [npm/cli#757 (comment)](https://github.com/npm/cli/issues/757#issuecomment-581773898)

@remusao thanks for digging! Well, in the past week I didn't think of any workaround for that, and cannot expect folks to run a hypothetically patched future version of node anyway, so I think I'll have to revert to webpack. Alternatives / ideas still welcome.

Reply:
Even if npm/cli#757 is fixed, webpack *might* still be needed...

My memory is a little hazy with this, but the main reason for the webpack workflow is because at that time, during `npm install -g nativefier`, the dependencies of the `./app` directory are placed in the users global `node_modules` (i.e. *not* `./app/node_modules`). 

When running electron packager on the `./app` directory, those dependencies declared in `./app/package.json` will not be packaged together with the application, as `./app/node_modules` is absent. Using webpack allows all the dependencies to be merged together into a single amalgamation in `./app/lib` to workaround this problem. See:
https://github.com/jiahaog/nativefier/blob/b927b401b0b953328ba1d880905b9d54be760265/gulp/build/build-app.js#L11

For some reason, I can find the `./app/node_modules` in the downloaded tarball of `7.7.1`, but this is not present in the older `6.11.0` version. Things might have changed on the npm side, or with one the of build scripts that I haven't dug into yet (I probably should do a bisect later), so this entire section above might not be an issue.

Let me think about this a little more, and get back to you.

Reply:
@jiahaog hmmm I don't know what in the past didn't work with `app/node_modules`, but today it works. Try the current build, you'll see that the `npm install` in `app` works, dependencies are installed as expected, then `electron-packager` does bundle `app` including `node_modules`, and resulting nativefier apps work perfectly.

Feel free to dig but, to me right now, this approach works, except for the npm 60s lock issue, which in my opinion is blocking. Maybe another _additional_ problem used to prevent this workflow in the past, but today it's no longer the case, as proved by the current state of the branch. I'm going to go ahead and restore webpacking the app into a single file.